core/src/hir/ty_position.rs

```diff
@@ -0,0 +1,28 @@
+use super::{Borrow, MaybeOwn, OutStructId, ReturnableStructPath, StructId, StructPath};
+use core::fmt::Debug;
+
+pub trait TyPosition: Debug + Copy {
+    const IS_OUT_ONLY: bool;
+    type OpaqueOwnership: Debug;
+    type StructId: Debug;
+    type StructPath: Debug;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct Everywhere;
+#[derive(Debug, Copy, Clone)]
+pub struct OutputOnly;
```

Comment:
I kind of wish that these names felt more like inverses. Some ideas:
* `FromEither` vs `FromRust`
* `TwoWay` vs `OneWay`
* `WithoutOwnership` vs `WithOwnership` (my favorite)

Since we're using type aliases most of the time, we wouldn't have to write out the names but I think it would be a lot clearer.

Reply:
Yeah i mean elsewhere we use OutputFoo and Foo so I stuck with that

Reply:
Would it be difficult to change things up? I'm actually really starting to like `TwoWay` and `OneWay`. But maybe that can be a separate PR after we have time to see how things feel.

Reply:
I think TwoWay and OneWay is potentially a problem if we introduce input-only types (we may have to!)

Yeah, this shoudl be discussed in a separate issue.